keps/sig-network/20191104-iptables-no-cluster-cidr.md

```diff
@@ -0,0 +1,358 @@
+capability by assuming that they can treat all the endpoints of a cluster as being equivalent and load balance service 
+traffic across all the endpoints, by just translating destination to the pod IP address.
+
+While this is powerful, it also means pod IP addresses are in many cases the constraining resource for cluster creation
+and scale. It would be valuable for implementations to have different strategies for managing pod IP addresses that can
+adapt to different environment needs.
+
+Some examples of use cases:
+
+   * Creating a cluster out of many disjoint ranges instead of a single range.
+   * Expanding a cluster with more disjoint ranges after initial creation.
+
+Not having to depend on the cluster pod CIDR for routing service traffic would effectively de-couple pod IP management
+and allocation strategies from service management and routing. Which in turn would mean that it would be far cheaper 
+to evolve the IP allocation schemes while sharing the same service implementation, thus significantly lowering the bar
+for adoption of alternate schemes.
+
+Alternate implementations that don’t use iptables could also adopt this same reasoning to not have to track the cluster
+CIDR for routing cluster traffic.
+
+### Goals
+
+   * Not having to depend on the cluster pod CIDR for iptable rules and cluster traffic routing.
+
+### Non-Goals
+
+   * Providing alternate models of IP allocation schemes for pod CIDR.
+   * Enhancing current allocators to handle disjoint ranges.
+   * Enhancing current allocators to add additional ranges after cluster creation.
+   * Changing current assumptions around having a single pod CIDR per node.
+
+## Proposal
+
+As stated above, the goal is to re-implement the functionality called out in the summary, but in a 
+way that does not depend on a pod cluster CIDR. The essence of the proposal is that for the 
+first two cases in iptables implementation and first case in ipvs, we can replace the `-s proxier.clusterCIDR` with 
+some notion of node local pod traffic.
+
+The core logic in these cases is “how to determine” cluster originated traffic from non-cluster originated ones. 
+The proposal is that tracking pod traffic generated from within the node is sufficient to determine cluster originated 
+traffic. For the first two use cases in iptables and first use case in ipvs, we provide alternatives to using 
+proxier.clusterCIDR in one of the following ways to determine cluster originated traffic
+
+   1. `-s node.podCIDR` (where node podCIDR is used for allocating pod IPs within the node)
+   2. `--in-interface prefix+` (where all pod interfaces start with same prefix,
+      or where all pod traffic appears to come from a single bridge or other interface)
+   3. `-m physdev --physdev-is-in` (for kubenet if we don’t want to depend on node podCIDR)
+
+Note the above are equivalent definitions, when considering only pod traffic originating from within the node.
```

Comment:
Ack, will add the flow for hostNetwork case.

Reply:
just to clarify, for hostNetwork, since the source IP will already be the node IP, the MASQUERADE action will be a no-op? 

Plus I don't see any rules today that seem to match by the detected node IP.

Reply:
added note that hostNetwork pods are not impacted by this change and should continue to work as is.

Reply:
Given the note, can this be marked as resolved? Should we add anything else?